### PR TITLE
feat: release 2.16.0

### PR DESCRIPTION
**Pull request by Baoyuantop**
Please answer these questions before submitting a pull request, **or your PR will get closed**.

**Why submit this pull request?**

- [ ] Bugfix
- [x] New feature provided
- [ ] Improve performance
- [ ] Backport patches

**What changes will this PR take into?**

Released version 2.16.0. It is basically compatible with APISIX 3.0.0.

**Checklist:**

- [x] Did you explain what problem does this PR solve? Or what new features have been added?
- [ ] Have you added corresponding test cases?
- [x] Have you modified the corresponding document?
- [x] Is this PR backward compatible? If it is not backward compatible, please discuss on the mailing list first

### CODE REVIEW

**Comment by juzhiyuan** [comment]
Please check the CI

**Comment by juzhiyuan** [comment]
Just to confirm: this PR doesn't need code changes?

**Comment by SkyeYoung** [comment]
Hi guys, I think we should add all PRs and issues related to this release to the PR description. 🤔 

(This may make it easier for people to understand what this PR has done.)

**Comment by bzp2010** [comment]
Actually I don't think this should be called 3.0.0, it's still a 2.x branch compatible version of APISIX 3.0, not the real 3.0. we'd better call it 2.16 or 2.99.

**Comment by Baoyuantop** [comment]
I agree with 2.16, what do you think?

File: README.md [hunk 1]
```diff
@@ -33,7 +33,7 @@
 
 - The master version should be used with Apache APISIX master version.
 
-- The latest released version is [2.15.0](https://apisix.apache.org/downloads/) and is compatible with [Apache APISIX 2.15.x](https://apisix.apache.org/downloads/).
+- The latest released version is [2.16.0](https://apisix.apache.org/downloads/) and is compatible with [Apache APISIX 3.0.x](https://apisix.apache.org/downloads/).
```
Comment (juzhiyuan) [inline README.md:36]:
```suggestion
- The latest released version is [2.16.0](https://apisix.apache.org/downloads/) and is compatible with [Apache APISIX 3.x](https://apisix.apache.org/downloads/).
```

**Review by juzhiyuan — APPROVED** [review]

**Review by guoqqqi — APPROVED** [review]

**Review by SkyeYoung — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -33,7 +33,7 @@
 
 - The master version should be used with Apache APISIX master version.
 
-- The latest released version is [2.15.0](https://apisix.apache.org/downloads/) and is compatible with [Apache APISIX 2.15.x](https://apisix.apache.org/downloads/).
+- The latest released version is [2.16.0](https://apisix.apache.org/downloads/) and is compatible with [Apache APISIX 3.x](https://apisix.apache.org/downloads/).
```
Comment (bzp2010) [inline README.md:37]: Given that the version numbers are currently no longer aligned with APISIX, I think we need to provide a compatibility list and we can do this in the next PR.

**Review by bzp2010 — APPROVED** [review]
LGTM, given that the version numbers are currently no longer aligned with APISIX, I think we need to provide a compatibility list and we can do this in the next PR. Just like:

**Comment by Baoyuantop** [comment]
> LGTM, given that the version numbers are currently no longer aligned with APISIX, I think we need to provide a compatibility list and we can do this in the next PR. Just like:
> 
> APISIX ver	APISIX Dashboard ver
> 3.0	2.16.x
> 2.15	2.15.x
> 2.14	2.14.x
> .....	.....

Agree

**Pull request merged**